scipy/linalg/tests/test_basic.py

```diff
@@ -771,13 +771,27 @@ def test_all_type_size_routine_combinations(self):
             elif assume_a == 'pos':
                 a = a.conj().T.dot(a) + 0.1*np.eye(size)
 
-            x = solve(a, b, assume_a=assume_a)
             tol = 1e-12 if dtype in (np.float64, np.complex128) else 1e-6
+
+            if assume_a in ['gen', 'sym', 'her']:
+                # We revert the tolerance from before
+                #   4b4a6e7c34fa4060533db38f9a819b98fa81476c
+                if dtype in (np.float32, np.complex64):
+                    tol *= 10
+
+            x = solve(a, b, assume_a=assume_a)
             assert_allclose(a.dot(x), b,
                             atol=tol * size,
                             rtol=tol * size,
                             err_msg=err_msg)
 
+            if assume_a == 'sym' and dtype not in (np.complex64, np.complex128):
+                x = solve(a, b, assume_a=assume_a, transposed=True)
+                assert_allclose(a.dot(x), b,
+                                atol=tol * size,
+                                rtol=tol * size,
+                                err_msg=err_msg)
+
 
```

Comment:
Could be useful to add a new test that does
```
a = np.eye(3).astype(complex)
with assert_raises(NotImplementedError):
    solve(a, a, transposed=True)
```

Reply:
I think this is also fixed.